### DISCUSSION TITLE
Is the jdk version not considered when trivy detects CVE-2022-22965?

### DISCUSSION THREAD

**Discussion by fighterhit** [aquasecurity/trivy#1929] [Q&A] [answered]
I have a question about the result of trivy scanning [CVE-2022-22965](https://github.com/advisories/GHSA-36p3-wjmg-h94x). According to the description of GitHub Advisory Database, this vulnerability requires jdk version greater than or equal to 9, but it seems that trivy does not take this into account even though jdk8 is actually being used?  There seem to be other users having the [same issue](https://github.com/aquasecurity/trivy/issues/1911#issuecomment-1086552776).

**Comment by DmitriyLewen** [comment, marked as answer]
Hello @fighterhit 
Trivy does not currently check the jdk version of your image.
Also, there is no field for the jdk version in the `GitHub Advisory Database` schema. 

That is why trivy finds [CVE-2022-22965](https://github.com/advisories/GHSA-36p3-wjmg-h94x) when you are using jdk8 or earlier.

You can use [.trivyignore](https://aquasecurity.github.io/trivy/v0.25.0/vulnerability/examples/filter/#by-vulnerability-ids) file to skip this vulnerability for jdk8. 

Regards, Dmitriy

**Reply by fighterhit** [reply]
Thank you @DmitriyLewen . When I read after the [.trivyignore](https://aquasecurity.github.io/trivy/v0.25.0/vulnerability/examples/filter/#by-vulnerability-ids) document, it seems that it requires to know the jdk version used by the container image in advance. However, since I use trivy to detect my users' image, I couldn't know the jdk version used in the image in advance.

**Reply by DmitriyLewen** [reply]
As far as I know, there is no file with information about used JDK version. We haven't way to get JDK version from the image.

**Reply by fighterhit** [reply]
Yes, maybe I have to execute `java -version` inside the container to get the jdk version.
